src/static/index.js

```diff
@@ -352,6 +333,54 @@ function convert_time(firebaseTime) {
     };
 }
 
+function loadConfiguration() {
+    unsubscribeConfigurationSnapshot && unsubscribeConfigurationSnapshot();
+    console.log('[debug] fetching configuration');
+    unsubscribeConfigurationSnapshot = db
+        .collection('configuration')
+        .onSnapshot(
+            (docs) => {
+                docs.forEach((doc) => {
+                    if (doc.id === 'referenceData') {
+                        const referenceData = doc.data();
+
+                        const tariffZones = JSON.parse(
+                            referenceData.tariffZones
+                        );
+                        app.ports.remoteConfigTariffZones.send(tariffZones);
```

Comment:
Lurer på om på et tidspunkt det kunne vært fint å ta en samtale på message passing og struktur på det via ports her. Blir veldig manuelt, verbose og sårbart for små kodefeil nå syns jeg med all den eksplisitte kodingen på dette. Nå som mer er over på Firestore kan det være interessant å integrere meldingsystemet der mer direkte med en slags dispatcher/CQRS style protokoll tettere integrert mot Firebase. Klart, ikke nødvendigvis helt dynamisk hvor det ikke er noe sikring på det som blir gjort, men litt mer strømlinjeformet og tydelige grenser. Altså egne union types med payload og action names i Elm f.eks. Det blir etterhvert litt sånn stor konseptuell/semantisk distanse mellom årsak og virkning uten tydelig kontrakter her, om du skjønner hva jeg mener.

Ikke noe actionable nå, men bare tenker høyt slik at det kan ev. trigge flere tanker av andre.

Reply:
Enig i at det er litt sårbart slik det er nå. Jeg så ikke klart for meg løsningsforslaget du presenterer, men om ikke annet skjønner jeg problemstillingen.

Reply:
Antar tanken er at Elm kan ta over denne biten litt mer, slik at i stedet for å hardkode masse ting i JavaScript-fila her (og en port for hver dokumenttype), så kan Elm-koden selv velge hva den vil spørre om, og man kunne hatt operasjoner som subscribe og unsubscribe fra Elm til JavaScript, og dokumenter og feiltilstander fra JavaScript til Elm.

Reply:
Yes. Pretty much